### PR TITLE
VideoPress: enqueue the VideoPress IFrame API asset file

### PR DESCRIPTION
**Pull request by retrofox**
Fixes #

## Proposed changes:

* VideoPress: enqueue the VideoPress IFrame API asset file

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Test in the front end and block editor contexts
* Open the network tab
* Filter JS files
* Filter by `iframe-api` 
* Confirm the file is loaded in the client

Screen Shot 2023-04-04 at 17 07 08

### CODE REVIEW

File: projects/packages/videopress/src/class-initializer.php [hunk 1]
```diff
@@ -228,6 +228,14 @@ public static function register_videopress_video_block() {
 			)
 		);
 
+		wp_enqueue_script(
+			self::JETPACK_VIDEOPRESS_VIDEO_VIEW_HANDLER . '-iframe-api',
+			'https://wordpress.com/wp-content/plugins/video/assets/js/videojs/videopress-iframe-api.js',
+			array(),
+			'1.0.0',
```
Comment (jeherve) [inline projects/packages/videopress/src/class-initializer.php]: Would it be worth updating this version number to something that would change every week, like we do for other resources loaded from WordPress.com? Something like for a flush every week for example:

```suggestion
			gmdate( 'YW' ),
```

I guess that depends how often this file will be changed.

File: projects/packages/videopress/src/class-initializer.php [hunk 2]
```diff
@@ -228,6 +228,14 @@ public static function register_videopress_video_block() {
 			)
 		);
 
+		wp_enqueue_script(
+			self::JETPACK_VIDEOPRESS_VIDEO_VIEW_HANDLER . '-iframe-api',
+			'https://wordpress.com/wp-content/plugins/video/assets/js/videojs/videopress-iframe-api.js',
```
Comment (jeherve) [inline projects/packages/videopress/src/class-initializer.php]: I believe this should use one of our static subdomains:

```suggestion
			'https://s0.wp.com/wp-content/plugins/video/assets/js/videojs/videopress-iframe-api.js',
```

**Review by jeherve — COMMENTED** [review]
Overall this works, but this introduces a second JavaScript file that's going to be loaded on every page of the site, whether it includes a video or not, just like in #24765. I think that's really something we should try to avoid, and instead we should aim to only include that file where it is needed.

File: projects/packages/videopress/src/class-initializer.php (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/class-initializer.php]: Yes, I think that lib won't be updated very often, but still. Thanks

**Comment by retrofox** [comment]
> Overall this works, but this introduces a second JavaScript file that's going to be loaded on every page of the site, whether it includes a video or not, just like in #24765. I think that's really something we should try to avoid, and instead we should aim to only include that file where it is needed.

We'll work on this asap in a follow-up one. Something that I tried to do, but I didn't how to do.
Two things to consider now:

* Enqueue (maybe) only in the front-end. Probably it will be used also in the block editor context
* Check whether the post content has some videopress/video block. If so, enqueue it

**Comment by retrofox** [comment]
> we should aim to only include that file where it is needed.

https://github.com/Automattic/jetpack/issues/29934

**Review by dhasilva — APPROVED** [review]

**Pull request merged**